Experiment automation/(LUMICKS) Reference kit automation/referencekit_automation_Bl_2_7_and_newer.py

```diff
@@ -0,0 +1,394 @@
+    """Move trap 1 until it reaches the `target` force on trap 2.
+
+    Parameters
+    ----------
+    target : float
+        The target force to reach [pN]
+    match_threshold: float
+        The threshold for the match score to check whether the beads are still caught [%]
+    speed : float
+        The speed at which to move the trap [um/s]
+    tolerance : float
+        The tolerance for the force measurement [pN]
+    tether_lost_threshold : float
+        The force threshold below which the tether is assumed lost [pN]
+
+    Raises
+    ------
+    RuntimeError
+        If the beads are lost (since we will not have a reliable distance then either)"""
+
+    df = target - force.latest_value
+    throw_if_beads_lost(match_threshold)
+
+    while abs(df) > tolerance:  # pN
+        bl.mirror1.move_by(dx=0.05 if df > 0 else -0.05, speed=speed)
+        df = target - get_force(dt=0.2)
+        throw_if_beads_lost(match_threshold)
+    check_tether_breakage(tether_lost_threshold)
+
+
+def get_force(dt=0.5):
+    t0 = bl.timeline.now
+    bl.pause(dt)
+    t1 = bl.timeline.now
+    return np.mean(force[t0:t1].data)
+
+
+def catch_dna(min_distance, max_distance, match_threshold, force_threshold, fishing_speed, fishing_attempts, dt, dna_length):
+    """Moves to the DNA channel and starts oscillating the trap until a prescribed force threshold is reached.
+
+    Parameters
+    ---------
+    min_distance : float
+        The minimum distance to oscillate the trap to when fishing for DNA (in microns)
+    max_distance : float
+        The maximum distance to oscillate the trap to (in microns)
+    match_threshold : float
+        The threshold for the match score to check whether the beads are still caught [%]
+    force_threshold : float
+        The force threshold to check whether we have caught DNA [pN]
+    fishing_speed : float
+        The speed at which to fish for DNA [um/s]
+    fishing_attempts : float
+        The number of attempts to fish for DNA
+    dt : float
+        The time interval over which the force is averaged [s]
+    """
+    print("Moving to DNA channel")
+    bl.microstage.move_to(name_dna_channel)
+    bl.pause(2.0)
+    print("reset force")
+    bl.reset_force()
+    bl.pause(1)
+    f_current = get_force(dt=dt)
+    attempts = 0
+    while f_current < force_threshold:
+        if attempts > fishing_attempts:
+            raise RuntimeError(f"Max DNA fishing attempts {fishing_attempts} reached.")
+
+        print(f"Fishing for DNA: attempt {attempts}/{fishing_attempts}")
+        goto_distance(min_distance, match_threshold, speed=fishing_speed)
+        bl.pause(0.5)
+        goto_distance(max_distance, match_threshold, speed=fishing_speed)
+        f_current = get_force(dt=dt)
+
+        attempts += 1
+
+
+def make_kymograph(force_kymo, match_threshold, name, path):
+    goto_force(force_kymo, match_threshold)
+    try:
+        goto_force(force_kymo, match_threshold)

```

Comment:
Why do you call this twice?
```suggestion
    try:
        goto_force(force_kymo, match_threshold)

```